mlflow/tracking/client.py

```diff
@@ -563,6 +567,78 @@ def delete_prompt(self, name: str, version: int):
         if not registry_client.get_latest_versions(name, stages=ALL_STAGES):
             registry_client.delete_registered_model(name)
 
+    # TODO: Use model_id in MLflow 3.0
+    @experimental
+    @require_prompt_registry
+    def log_prompt(self, run_id: str, prompt_uri: str) -> None:
+        """
+        Associate a prompt registered within the MLflow Prompt Registry with an MLflow Run.
+
+        Args:
+            run_id: The ID of the run to log the prompt to.
+            prompt_uri: The prompt URI in the format "prompts:/name/version".
+        """
+        prompt = self.load_prompt(prompt_uri)
+        if run_id_tags := prompt._tags.get(PROMPT_ASSOCIATED_RUN_IDS_TAG_KEY):
+            run_ids = run_id_tags.split(",")
+            run_ids.append(run_id)
+        else:
+            run_ids = [run_id]
+
+        name, version = parse_prompt_uri(prompt_uri)
+        self.set_model_version_tag(
+            name, version, PROMPT_ASSOCIATED_RUN_IDS_TAG_KEY, ",".join(run_ids)
+        )
+
+    # TODO: Use model_id in MLflow 3.0
+    @experimental
+    @require_prompt_registry
+    def detach_prompt_from_run(self, run_id: str, prompt_uri: str) -> None:
+        """
+        Detach a prompt registered within the MLflow Prompt Registry from an MLflow Run.
+
+        Args:
+            run_id: The ID of the run to log the prompt to.
+            prompt_uri: The prompt URI in the format "prompts:/name/version".
+        """
+        prompt = self.load_prompt(prompt_uri)
+        run_id_tags = prompt._tags.get(PROMPT_ASSOCIATED_RUN_IDS_TAG_KEY, "")
+        run_ids = run_id_tags.split(",")
+
+        if run_id not in run_ids:
+            raise MlflowException(
+                f"Run '{run_id}' is not associated with prompt '{prompt_uri}'.",
+                INVALID_PARAMETER_VALUE,
+            )
+
+        run_ids.remove(run_id)
+
+        name, version = parse_prompt_uri(prompt_uri)
+        self.set_model_version_tag(
+            name, version, PROMPT_ASSOCIATED_RUN_IDS_TAG_KEY, ",".join(run_ids)
+        )
+
+    # TODO: Use model_id in MLflow 3.0
+    @experimental
+    @require_prompt_registry
+    def list_logged_prompts(self, run_id: str) -> list[Prompt]:
```

Comment:
```suggestion
    def list_prompts(self, run_id: str) -> list[Prompt]:
```

Do we need `logged_`?

Reply:
Yea I would like to have a name with explicit mention that "this only list prompts *logged* to a particular Run", rather than all *registered* prompts.